subcommands/keys/ca_create.go

```diff
@@ -0,0 +1,126 @@
+package keys
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/foundriesio/fioctl/subcommands"
+)
+
+var (
+	createOnlineCA bool
+	createLocalCA  bool
+)
+
+func init() {
+	cmd := &cobra.Command{
+		Use:   "create <PKI Directory>",
+		Short: "Create PKI infrastructure to manage mutual TLS for the device gateway",
+		Run:   doCreateCA,
+		Args:  cobra.ExactArgs(1),
+		Long: `Peform a one-time operation to set up PKI infrastructure for managing
+the device gateway. This command creates a few things:
+
+# Root of trust for your factory: factory_ca.key / factory_ca.pem
+The factory_ca keypair is generated by this command to define the PKI root of
+trust for this factory.
+ * factory_ca.key - This is a private key that should be STORED OFFLINE.
```

Comment:
Should we also give a bit more details on what type of key this really is? Usage only says private key.

Didn't yet check the content of the create_ca script, but would it be possible later to also leverage PKCS11/HSM here as well? This would be the key for the root CA, so we can expect customers to use a HSM to manage that.

Reply:
I should add the key types. this could leverage hsm, but has no support at the moment unless you manually do the steps performed by these scripts:

 https://git.foundries.io/development/cloudplatforms/infrastructure/ota-lite/tree/master/ota_lite/templates

I had this idea when building this feature that the scripts living here would be handy. They drive the unit-tests, on-prem, and also fioctl. So you have one thing that gets lots of coverage. However, I do wonder if maybe we should do this key management logic directly in fioctl? Or maybe, do it like this for now, but move it all into fioctl down the road?

Reply:
force push explains its an "EC prime256v1" key

Reply:
@ricardosalveti - i just force pushed support for HSM. Let me know what you think. The corresponding backend code is here:
 https://git.foundries.io/development/cloudplatforms/infrastructure/ota-lite/merge_requests/89